e2xgrader/utils/mode.py

```diff
@@ -0,0 +1,128 @@
+import os
+from typing import Dict
+
+from jupyter_core.paths import jupyter_config_path
+
+from .. import _jupyter_server_extension_paths
+from ..extensions.utils import discover_nbextensions, get_notebook_config_manager
+
+
+def get_nbextension_config() -> Dict[str, Dict[str, bool]]:
+    """
+    Get the configuration for nbextensions.
+
+    Returns:
+        A dictionary containing the configuration for nbextensions.
+        The dictionary has the following structure:
+        {
```

Comment:
why twice? what's the diff between tree and notebook?

Reply:
Because we have nbextensions that are visible in the tree view (File view, like how in student_exam mode the delete notebook UI is gone) and nbextensions that are visible in a notebook (Notebook view, like the cells). This is more for sanity checking that not only notebook or only tree extensions are active.